src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,107 @@
+/*
+    Jon Kaimmer
+    CS410 Software Engineering
+    Spring 2023
+    Junit Testing Assignment
+
+    Spring Manipulation
+ */
+
+import java.lang.IllegalArgumentException;
+import java.lang.IndexOutOfBoundsException;
+import java.lang.NullPointerException;
+import java.util.*;
+
+
 public class StringManipulation implements StringManipulationInterface {
 
+    public String s;
```

Comment:
Changing String from public to protected/private here would increase code encapsulation.